### PR TITLE
WT-9477 Don't allow  checkpoint triggering internal page split which leads to corrupted internal page in the checkpoint

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by agorrod** [comment]
We are going to hold off on this while we build a test program that exercises the code.

**Review by agorrod — CHANGES_REQUESTED** [review]
I have suggested an edit to the comment (thanks for adding one).

File: src/btree/bt_split.c [hunk 1]
```diff
@@ -1271,8 +1271,12 @@ __split_parent_climb(WT_SESSION_IMPL *session, WT_PAGE *page)
      * to a different part of the tree where it will be written; in other words, in one part of the
      * tree we'll skip the newly created insert split chunk, but we'll write it upon finding it in a
      * different part of the tree.
+     *
+     * We don't allow checkpoint to trigger internal split as well because it may confuse itself to
+     * write the old disk image for the splitting internal page resulting to some leaf pages being
+     * referenced by two internal pages in the checkpoint.
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply src/btree/bt_split.c]: [345c2e4](https://github.com/wiredtiger/wiredtiger/pull/8046/commits/345c2e4458a831dbc4c59569283d6580d9bbc2d7)

**Review by agorrod — APPROVED** [review]
LGTM if it passes testing.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**